packages/cozy-app-publish/cozy-app-publish.js

```diff
@@ -107,45 +114,35 @@ function _getPublishMode() {
 
 async function publishApp(cliOptions) {
   const publishMode = _getPublishMode()
-  if (publishMode === MODES.TRAVIS) {
-    console.log()
-    console.log(`${colorize.bold('Travis')} ${colorize.blue('publish mode')}`)
-    console.log()
-    try {
-      await scripts.travis({
-        registryToken: cliOptions.token,
-        buildDir: cliOptions.buildDir,
-        buildCommit: cliOptions.buildCommit,
-        buildUrl: cliOptions.buildUrl,
-        prepublishHook: cliOptions.prepublishHook,
-        postpublishHook: cliOptions.postpublishHook,
-        registryUrl: cliOptions.registryUrl,
-        spaceName: cliOptions.space,
-        verbose: cliOptions.verbose
-      })
-    } catch (error) {
-      console.error(`↳ ❌  ${error.message}`)
-      process.exit(1)
-    }
-  } else if (publishMode === MODES.MANUAL) {
-    console.log()
-    console.log(`${colorize.bold('Manual')} ${colorize.blue('publish mode')}`)
-    console.log()
-    scripts.manual({
-      registryToken: cliOptions.token,
-      buildDir: cliOptions.buildDir,
+  const acceptedModes = Object.values(MODES).includes(publishMode)
+  if (!acceptedModes.includes(publishMode)) {
+    throw new Error(
+      `Absent or unrecognized mode, you passed ${publishMode}. Accepted modes: ${acceptedModes}.`
+    )
+  }
+
+  const publishFn = scripts[publishMode]
+
+  console.log()
+  console.log(
+    `${colorize.bold(capitalize(publishMode))} ${colorize.blue('publish mode')}`
+  )
+  console.log()
+  return publishFn(
```

Comment:
We're now missing the `try/catch` block around this call (https://github.com/cozy/cozy-libs/pull/327/files#diff-18ba38e9e92d14c179d04b2d7403e1cbL127)

Reply:
Yes, this is already taken care of here https://github.com/cozy/cozy-libs/blob/985908611148b58507430c99d5325a99f61f2764/packages/cozy-app-publish/cozy-app-publish.js#L87

Reply:
Ah yes, allright!